### PR TITLE
fix: serve missing logic from #3321

### PR DESCRIPTION
**Pull request by aarnphm**
Signed-off-by: Aaron Pham <29749331+aarnphm@users.noreply.github.com>

### CODE REVIEW

File: src/bentoml/_internal/server/server.py [hunk 1]
```diff
@@ -1,31 +1,58 @@
...
+        self,
+        exc_type: type[BaseException],
+        exc_value: BaseException,
+        traceback_type: TracebackType,
+    ):
+        try:
+            self.stop()
+        except Exception as e:  # pylint: disable=broad-except
+            logger.error(f"Error stopping server: {e}", exc_info=e)
+        traceback.print_exception(exc_type, exc_value, traceback_type)
```
Comment (sauyon) [inline src/bentoml/_internal/server/server.py:58]: Should this be in the except block?

File: src/bentoml/bentos.py [hunk 2]
```diff
@@ -427,76 +427,86 @@ def containerize(bento_tag: Tag | str, **kwargs: t.Any) -> bool:
...
+        if host is None:
+            host = BentoMLContainer.http.host.get()
+        if port is None:
+            port = BentoMLContainer.http.port.get()
+            ssl_args.update(
+                ssl_keyfile_password=ssl_keyfile_password,
+                ssl_version=ssl_version,
+                ssl_cert_reqs=ssl_cert_reqs,
+                ssl_ciphers=ssl_ciphers,
+            )
```
Comment (sauyon) [inline src/bentoml/bentos.py:473]: Er, why is this inside `if port is None`?

File: src/bentoml/bentos.py [hunk 3]
```diff
@@ -427,76 +427,86 @@ def containerize(bento_tag: Tag | str, **kwargs: t.Any) -> bool:
...
         "--backlog",
         str(backlog),
+        *construct_ssl_args(**ssl_args),
     ]
     if production:
         args.append("--production")
-    if reload:
-        args.extend(["--reload", str(reload)])
+    elif reload:
+        args.append("--reload")
```
Comment (sauyon) [inline src/bentoml/bentos.py:499]: Let's change this back; if the user specifies both we should let the CLI warn them.

File: src/bentoml/bentos.py (hunk 2 above)
Comment (aarnphm) [reply src/bentoml/bentos.py:473]: 😮‍💨 oopsie

**Review by sauyon — APPROVED** [review]

**Pull request merged**